File: chapter-1/exercises/python/binarySearch.py
def binary_search(list, item):
	low = 0
	high = len(list) - 1

	while low <= high:
		mid = (low + high) // 2
		guess = list[mid]
		if guess == item:
			return mid
		if guess > item:
			high = mid - 1
		else:
			low = mid + 1
	return None

def createList(r1, r2):
    return list(range(r1, r2+1))

my_list = createList(0, 128)

print(binary_search(my_list, -2))

File: chapter-1/exercises/javascript/binarySearch.js
function binarySearch(array, target) {
	let low = 0
	let high = array.length - 1

	while (low <= high) {
		let mid = parseInt((low + high) / 2)
		let guess = array[mid]

		if (guess === target) {
			return mid
		}

		if (guess > target) {
			high = mid - 1
		} else {
			low = mid + 1
		}
	}
	return null
}

const arr = [1, 3, 4, 5, 7, 8, 10, 12, 13, 15]

console.log(binarySearch(arr, 16))
